### PR TITLE
Datastore & Plugin: Remove active port details when volume is deleted

### LINKED ISSUE #271: Volume delete does not remove port details leading to consumed port error.
When a volume is deleted currently it does not remove the port consumed by storage_units after the command is called, and port related data in `ports` table is intact. 
This leads to consumed port / port already in use error when a new volume is to be created in the same port again. 

Solution: 
Delete port from ports table for the corresponding pool. node before volume delete is called.

### PR DESCRIPTION
**Pull request by vatsa287**
Fixes: #271
Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>

### CODE REVIEW

File: mgr/src/server/datastore/ports.cr [hunk 1]
```diff
@@ -39,4 +39,15 @@ module Datastore
     query = "DELETE FROM ports WHERE pool_id = ? AND node_id = ? AND updated_on < datetime('now', '-5 minutes')"
     connection.exec(query, pool_id, node_id)
   end
+
+  def delete_port(pool_id : String, distribute_groups : Array)
+    distribute_groups.each do |dist_grp|
+      dist_grp.storage_units.each do |storage_unit|
+        query = "DELETE FROM ports WHERE pool_id = ? AND node_id = ? AND port = ?"
+        puts "pool_id: #{pool_id}, node_id: #{storage_unit.node.id}, port: #{storage_unit.port}"
```
Comment (aravindavk) [inline mgr/src/server/datastore/ports.cr]: Remove this line

File: mgr/src/server/datastore/ports.cr [hunk 2]
```diff
@@ -39,4 +39,15 @@ module Datastore
     query = "DELETE FROM ports WHERE pool_id = ? AND node_id = ? AND updated_on < datetime('now', '-5 minutes')"
     connection.exec(query, pool_id, node_id)
   end
+
+  def delete_port(pool_id : String, distribute_groups : Array)
```
Comment (aravindavk) [inline mgr/src/server/datastore/ports.cr]: change name to `delete_reserved_volume_ports`

**Review by aravindavk — COMMENTED** [review]
Add test cases:

```ruby
TEST "kadalu volume create dev/vol_port1 server1:5001:/data/vol_port/s1 --no-start"
TEST "kadalu volume delete dev/vol_port1"
TEST "kadalu volume create dev/vol_port2 server1:5001:/data/vol_port/s1 --no-start"
TEST "kadalu volume delete dev/vol_port2"
```

File: mgr/src/server/datastore/ports.cr (hunk 1 above)
Comment (vatsa287) [reply mgr/src/server/datastore/ports.cr]: Ack

File: mgr/src/server/datastore/ports.cr (hunk 2 above)
Comment (vatsa287) [reply mgr/src/server/datastore/ports.cr]: Done

**Review by aravindavk — APPROVED** [review]

**Pull request merged**